src/components/shared/Swiper.tsx

```diff
@@ -0,0 +1,329 @@
+    this.setState({
+      offset: offset
+    });
+    // Update index in the state
+    this.setState({
+      index
+    });
+  };
+
+  /**
+   * Swipe one slide forward
+   */
+  swipe = () => {
+    // Ignore if already scrolling or if there is less than 2 slides
+    if (this.state.isScrolling || this.state.total < 2) {
+      return;
+    }
+
+    const state = this.state,
+      diff = this.state.index + 1,
+      x = diff * state.width,
+      y = 0;
+
+    // Call scrollTo on scrollView component to perform the swipe
+    this.scrollView && this.scrollView.scrollTo({ x, y, animated: true });
+
+    // Update internal scroll state
+    this.setState({
+      isScrolling: true
+    });
+
+    // Trigger onScrollEnd manually on android
+    // TODO: Revisit for Android https://github.com/facebook/react-native/issues/11693
+    // if (Platform.OS === "android") {
+    //   setImmediate(() => {
+    //     this.onScrollEnd({
+    //       nativeEvent: {
+    //         position: diff
+    //       }
+    //     });
+    //   });
+    // }
+  };
+
+  /**
+   * Render ScrollView component
+   * @param {array} slides to swipe through
+   */
+  renderScrollView = (pages: React.ReactNode[]) => {
+    return (
+      <ScrollView
+        ref={component => {
+          this.scrollView = component;
+        }}
+        {...this.props}
+        onScrollBeginDrag={this.onScrollBegin}
+        onMomentumScrollEnd={this.onScrollEnd}
+        onScrollEndDrag={this.onScrollEndDrag}
+      >
+        {pages.map((page, i) => (
+          // Render each slide inside a View
+          <View style={[styles.fullScreen, styles.slide]} key={i}>
+            {page}
+          </View>
+        ))}
+      </ScrollView>
+    );
+  };
+
+  /**
+   * Render pagination indicators
+   */
+  renderPagination = () => {
+    if (this.state.total <= 1) {
+      return null;
+    }
+
+    const ActiveDot = <View style={[styles.dot, styles.activeDot]} />,
+      Dot = <View style={styles.dot} />;
+
+    let dots = [];
+
+    for (let key = 0; key < this.state.total; key++) {
+      dots.push(
+        key === this.state.index
+          ? // Active dot
+            React.cloneElement(ActiveDot, { key })
+          : // Other dots
+            React.cloneElement(Dot, { key })
+      );
+    }
+
+    return (
+      <View pointerEvents="none" style={[styles.pagination, styles.fullScreen]}>
+        {dots}
+      </View>
+    );
+  };
+
+  /**
+   * Render the component
+   */
+  render() {
+    return (
+      <View style={[styles.container, styles.fullScreen]}>
+        {this.renderScrollView(this.props.children)}
+        {this.renderPagination()}
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  fullScreen: {
+    width: width,
+    height: height
+  },
+  container: {
+    backgroundColor: "transparent",
+    position: "relative"
+  },
+  slide: {
+    backgroundColor: "transparent"
+  },
+  pagination: {
+    position: "absolute",
+    bottom: 110,
```

Comment:
I wonder if we really need to do this kind of absolute positioning, but it's not important enough right now to really be a big deal

Reply:
Nope, it doesn't need to be. Added the issue here https://github.com/rahafoundation/raha-mobile-app/issues/14